src/modules/Game.class.js

```diff
@@ -20,25 +17,97 @@
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
     // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.state = 'idle';
+    this.score = 0;
+    this.initialState = initialState;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    if (this.state !== 'idle') {
+      const moveAllNonZeroToTheLeft = this.initialState.map((row) =>
+        mergeToTheLeft(row),
+      );
+
+      this.state = loseWinState(moveAllNonZeroToTheLeft);
+
+      this.score += calculateScore(this.initialState, moveAllNonZeroToTheLeft);
+
+      const newCellAdded = addNewCellRandomCell(moveAllNonZeroToTheLeft, 1);
+
+      this.initialState = newCellAdded;
+
+      return this.initialState;
+    }
+  }
+  moveRight() {
+    if (this.state !== 'idle') {
+      const moveAllNonZeroToTheRight = this.initialState.map((row) =>
+        mergeToTheRight(row),
+      );
+
+      this.state = loseWinState(moveAllNonZeroToTheRight);
+
+      this.score += calculateScore(this.initialState, moveAllNonZeroToTheRight);
+
+      const newCellAdded = addNewCellRandomCell(moveAllNonZeroToTheRight, 1);
+
+      this.initialState = newCellAdded;
+
+      return this.initialState;
+    }
+  }
+  moveUp() {
+    if (this.state !== 'idle') {
+      const moveAllNonZeroToTheUp = mergeUp(this.initialState);
+
+      this.state = loseWinState(moveAllNonZeroToTheUp);
+
+      this.score += calculateScore(this.initialState, moveAllNonZeroToTheUp);
+
+      const newCellAdded = addNewCellRandomCell(moveAllNonZeroToTheUp, 1);
+
+      this.initialState = newCellAdded;
+
+      return this.initialState;
+    }
+  }
+  moveDown() {
+    if (this.state !== 'idle') {
+      const moveAllNonZeroToTheDown = mergeDown(this.initialState);
+
+      this.score += calculateScore(this.initialState, moveAllNonZeroToTheDown);
+      this.state = loseWinState(moveAllNonZeroToTheDown);
```

Comment:
The order of updating the state and score should be consistent across all move methods. Consider updating the state after calculating the score.